evap/evaluation/templates/base.html

```diff
@@ -85,41 +83,70 @@
+                        element.parentElement.classList.remove("is-invalid");
+                        element.classList.add("is-invalid");
+                    }
+                    const baseOptions = {
+                        createOnBlur: true,
                         placeholder: "{% trans 'Please select...' %}",
-                        minimumResultsForSearch: 15,
-                        minimumInputLength: $(this).children('option').length > 50 ? 3 : 0,
-                        allowClear: this.multiple,
-                        dropdownParent: parent
-                    });
+                        hidePlaceholder: true,
+                        render: {
+                            option_create: function( data, escape ){return '<div class="create">' + escape(data.input) + '</div>';},
+                            no_results: function( data, escape ){return '<div class="no-results">{% trans "No results found" %}</div>';},
+                        },
+                        plugins: {
+                            // "dropdown_input": {}, // # TODO: Do we want this? I think both are a bit weird, with and without
+                        },
+                    };
+                    if(element.hasAttribute("data-tomselect-fullwidth")) {
+                        // span needed to the "remove this icon" button is right-aligned
+                        baseOptions.render.item = function(data, escape) {return '<div class="w-100"><span class="w-100">' + escape(data.text) + '</span></div>';};
+                    }
+                    if(element.multiple) {
+                        baseOptions.plugins.clear_button = {"title": "{% trans 'Remove all items' %}"};
+                        baseOptions.plugins.remove_button = {"title": "{% trans 'Remove this item' %}"};
+                    }
+                    if(element.getElementsByTagName("option").length > 50) {
+                        // # TODO: This doesn't work
+                        baseOptions.shouldLoad = function(query) {return query.length >= 3;};
+                    }
+                    new TomSelect(element, Object.assign({}, baseOptions, additionalOptions));
                 });
             };
-            applySelect2($('select').not('.form-template select').not('[disabled]').not('.no-select2').not('#id_delegate_to'));
-            // handle delegation modal, which includes a dropdown select field
-            applySelect2($('select#id_delegate_to'), $('#delegateSelectionModal'));
+            applyTomSelect(document.querySelectorAll("select:not(.form-template select):not(.no-tomselect)"));
 
             // don't use the placeholder text on disabled select2 elements
-            applySelect2Disabled = function(elem) {
-                elem.select2();
-            };
-            applySelect2Disabled($('select[disabled]').not('.form-template select'));
+            // # TODO
+            // applySelect2Disabled = function(elem) {
```

Comment:
Where on the page do we have disabled select fields that have no value set and thus would show the placeholder? And why do we want to handle them different? Isn't a disabled "Please select..." pretty clear while a disabled field with no contents looks like a frontend error?

Reply:
> Where on the page do we have disabled select fields that have no value set and thus would show the placeholder? 

@janno42 do you know of such a disabled field?

Reply:
i can't think of such a field right now. it's probably okay to have the placeholder text in disabled fields as well.